### PR TITLE
Block deletion of CCS secret

### PR DESCRIPTION
**Pull request by NautiluX**
### What type of PR is this? 
_bug_

### Which issue(s) this PR fixes(can be a reference to existing issue or jira/bz):

_Fixes OSD-5361_

### Pre-checks:
- [x] manually tested latest changes against crc/k8s
- [x] run the `make coverage` command to generate new calculated coverage

### CODE REVIEW

**Review by georgettica — COMMENTED** [review]
looks fine, untested on another environments

small nits

File: pkg/controller/projectclaim/projectclaim_controller_test.go [hunk 1]
```diff
@@ -109,13 +109,30 @@ var _ = Describe("ProjectclaimController", func() {
...
+
+					Context("When it's not a CCS cluster or owner reference is set", func() {
+						BeforeEach(func() {
+							mockAdapter.EXPECT().EnsureCCSSecretOwnerReference().Return(gcputil.ContinueProcessing())
+						})
+						It("Sets the state to PendingProject", func() {
+							mockAdapter.EXPECT().EnsureProjectClaimStatePendingProject()
+							res, err := reconciler.ReconcileHandler(mockAdapter)
+							Expect(err).ToNot(HaveOccurred())
+							Expect(res.Requeue).To(Equal(false))
```
[the remark this answers is not in the document]
Comment (NautiluX) [reply pkg/controller/projectclaim/projectclaim_controller_test.go:131]: no worky, this is testing Reconcile.

File: pkg/controller/projectclaim/projectclaim_controller_test.go [hunk 2]
```diff
@@ -109,13 +109,30 @@ var _ = Describe("ProjectclaimController", func() {
...
+							Expect(res.RequeueAfter).To(Equal(0 * time.Second))
+						})
+
+					})
+
+					Context("When it's not a CCS cluster or owner reference is set", func() {
+						BeforeEach(func() {
+							mockAdapter.EXPECT().EnsureCCSSecretOwnerReference().Return(gcputil.ContinueProcessing())
+						})
+						It("Sets the state to PendingProject", func() {
```
[the remark this answers is not in the document]
Comment (NautiluX) [reply pkg/controller/projectclaim/projectclaim_controller_test.go:127]: this test was already there, it just moved to a different context

File: pkg/controller/projectclaim/projectclaim_controller_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (georgettica) [reply pkg/controller/projectclaim/projectclaim_controller_test.go:131]: 👍

File: pkg/controller/projectclaim/projectclaim_controller_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (georgettica) [reply pkg/controller/projectclaim/projectclaim_controller_test.go:127]: so the naming is incorrect and should be changed?

File: pkg/controller/projectclaim/projectclaim_controller_test.go [hunk 3]
```diff
@@ -109,13 +109,30 @@ var _ = Describe("ProjectclaimController", func() {
 						mockAdapter.EXPECT().EnsureFinalizer().Return(gcputil.ContinueProcessing())
 					})
 
-					It("Sets the state to PendingProject", func() {
```
[the remark this answers is not in the document]
Comment (NautiluX) [reply pkg/controller/projectclaim/projectclaim_controller_test.go:112]: It's not, it just moved to a different project

**Review by sam-nguyen7 — COMMENTED** [review]
@rafael-azevedo and I looked this over. This will not prevent the CCS secret from being deleted when the cluster namespace is being deleted. In order to do so, we need to add a finalizer to the CCS secret then remove it when the projectclaim is deleted if it is CCS.

It is still a good idea to have a CCS secret owner reference because this will ensure the CCS secret is deleted when the projectClaim is deleted.

**Comment by NautiluX** [comment]
This won't work. Misunderstood owner reference, we need to use finalizers instead, will open a separate PR for it.

**Comment by NautiluX** [comment]
> @rafael-azevedo and I looked this over. This will not prevent the CCS secret from being deleted when the cluster namespace is being deleted. In order to do so, we need to add a finalizer to the CCS secret then remove it when the projectclaim is deleted if it is CCS.
> 
> It is still a good idea to have a CCS secret owner reference because this will ensure the CCS secret is deleted when the projectClaim is deleted.

As far as I understand owner references, it's the other way round. When the secret is deleted, the ProjectClaim would be deleted as well. But, the secret won't stay until the ProjectClaim is deleted, as the default `PropagationPolicy` is not `Foreground` and this needs to be set during the deletion api call.

**Pull request closed without merging**